docs: Add missing sign in resolver step in getting started

OrkoHunter opened the pull request:
## Hey, I just made a Pull Request!

<!-- Please describe what you added, and add a screenshot if possible.
     That makes it easier to understand the change so we can :shipit: faster. -->

Since v1.1.0, configuring a sign-in resolver is mandatory. So, this PR adds the missing step in getting started.

This PR also updates the code examples in identity resolver page for better understanding of how to use them.


#### :heavy_check_mark: Checklist

<!--- Please include the following in your Pull Request when applicable: -->

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

freben commented on docs/getting-started/configuration.md:
```diff
@@ -187,10 +189,77 @@ components: {
 
+### Add sign-in resolver in the backend
+
+The Auth backend plugin is responsible for assigning a [Backstage User Identity](../auth/identity-resolver.md#backstage-user-identity) when a user signs in. The identity is mainly used in determining the ownership of Backstage Catalog entities for the user. This is achieved by writing a [Sign In Resolver](../auth/identity-resolver.md#sign-in-resolvers) which takes care of assigning the right iden ...
```
The second sentence could be adjusted a bit. The identity part of it (the `sub`) is your identity and that identifies your user for any purpose within the Backstage ecosystem. It can for example be used as a primary key in a per-user settings backend, or any other similar purpose. The `ent` is what relates to ownership (the list of entity refs through which you claim ownership).

OrkoHunter replied:
Hmm, I was afraid of being too technical in this line and give a one-liner for them to read and move on with auth ready. The Sign In Resolver page talks about `sub` and `ent` in good detail.

But let me adjust here per your suggestion!

freben commented on docs/auth/identity-resolver.md:
```diff
@@ -164,40 +175,49 @@ that happens during sign-in you can replace `ctx.signInWithCatalogUser` with a s
+        // the standard logic. It uses a reference to the entity itself, as well as the
+        // target of each `memberOf` relation where the target is of the kind `Group`.
+        //
+        // If you replace the catalog lookup with something does not return
```
```suggestion
        // If you replace the catalog lookup with something that does not return
```

freben commented on docs/getting-started/configuration.md:
```diff
@@ -187,10 +189,77 @@ components: {
+            }
+            // Split the email into the local part and the domain.
+            // You can use the email domain and verify that it belongs to your own org
+            // e.g. acme.org. It is recommended to include this kind of check for security.
```
since it's so strongly recommended, I think it's worth doing the actual check on the next line (as in the earlier example above).

freben reviewed (commented):
Thanks!

Rugvip commented on docs/getting-started/configuration.md:
```diff
@@ -187,10 +189,77 @@ components: {
 
-That should be it. You can stop your Backstage App. When you start it again and
-go to your Backstage portal in your browser, you should have your login prompt!
+For now, let's create a simple Sign In Resolver which uses the first part of the user's email address to assign a Backstage User Identity.
```
We actually just identified as part of the setup simplification work that we should have a very simple guest sign-in resolver. One that treats every single user that sign in as the same "guest" identity.

What do you think about adding that to its own section in `../auth/identity-resolver.md`, and then linking to that section from here? Kinda hate maintaining example codeblocks for the same thing in multiple places. We keep seeing again and again that they start to diverge and end up outdated.

Rugvip replied:
Here it is #11937 :grin:

OrkoHunter replied:
Amazing! @Rugvip When do you think https://github.com/backstage/backstage/pull/11937 is going to be released?

Let me update this PR with a note that you don't need to do anything with sign in resolvers as getting started as it comes with a guest resolver.

Rugvip replied:
Gonna be released tomorrow :grin:

OrkoHunter commented:
@freben Addressed your comments - thank you! :)
@Rugvip The default sign in resolver would be awesome, absolutely! Until it is released, we should ensure that our getting started page isn't broken. It's include without an easy to use sign in resolver. So, we can replace this custom GitHub resolver that is being added in this PR with the default guest resolver when it's available!

Rugvip commented on .github/vale/Vocab/Backstage/accept.txt:
```diff
@@ -92,6 +92,7 @@ elasticsearch
 esbuild
 eslint
 etag
+ent
```
Not needed anymore right? This should always be wrapped in a code block anyway so let's avoid adding it.

Rugvip reviewed (approved):
Alright, guest resolver update is now shipped, so let's :shipit:!

Just a tiny tweak
